### PR TITLE
feat: Add utility methods to help conversion between literals.

### LINKED ISSUE #33: Allow conversion between literal values
I noticed the following: https://github.com/apache/iceberg-rust/pull/29/files#diff-a129ecbb5292628eaa22ebd56d7e869efaaf05f27646c7e724a205b4564505abR745

```rust
Uuid::parse_str("0db3e2a8-9d1d-42b9-aa7b-74ebe558dceb").unwrap()
```

I would do a suggestion here that we're using in PyIceberg and that works very well.

```python
literal("0db3e2a8-9d1d-42b9-aa7b-74ebe558dceb").to(UuidType())
literal("true").to(BooleanType())
literal("2017-08-18"), DateType()
```

Where the `literal("true")` produces a `StringLiteral`, and then we can convert it to different literal types. 

It also works when starting from other types:
```python
literal(34).to(FloatType())
# Days since 1970-01-01
literal(12345).to(DateType())
```

This way we can accept any value, and when binding it to a field, convert it to the correct type. This works very well in Java and PyIceberg when accepting an expression `timestamp >= '2021-01-01'` where the date is first a string, and then we can convert this to the type of `timestamp` when binding it to the schema a timestamp in this case. Similarly, if the table has a `date(timestamp)` partition, we can convert it to a `DateType` to compare it to the partition. If the table would evolve its partitioning scheme from `date` to `hour`, we could just convert it to a `datetime` (miliseconds since epoch) and use this for the new partitioning scheme.

### PR DESCRIPTION
**Pull request by liurenjie1024**
Close #33 .

Thanks for @Fokko 's suggestion. Add them so that we can make literals more convenient.

### CODE REVIEW

File: crates/iceberg/src/spec/values.rs [hunk 1]
```diff
@@ -86,6 +88,349 @@ pub enum Literal {
...
+    /// Example
+    /// ```rust
+    /// use iceberg::spec::values::Literal;
+    /// let t = Literal::date_from_str("1970-01-03").unwrap();
+    ///
+    /// assert_eq!(Literal::date(2), t);
+    /// ```
+    pub fn date_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
+        let t = s.as_ref().parse::<NaiveDate>().map_err(|e| {
+            Error::new(ErrorKind::DataInvalid, "Can't parse date".to_string()).with_source(e)
```
Comment (Xuanwo) [inline crates/iceberg/src/spec/values.rs:205]: It's better to print the string, like `Can't parse '{s}' as date`.

File: crates/iceberg/src/spec/values.rs [hunk 2]
```diff
@@ -86,6 +88,349 @@ pub enum Literal {
...
+    ///     1 * 3600 * 1_000_000 + // 1 hour
+    ///     2 * 60 * 1_000_000 +   // 2 minutes
+    ///     1 * 1_000_000 + // 1 second
+    ///     888999  // microseconds
+    /// };
+    /// assert_eq!(Literal::time(micro_secs), t);
+    /// ```
+    pub fn time_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
+        let t = s.as_ref().parse::<NaiveTime>().map_err(|e| {
+            Error::new(ErrorKind::DataInvalid, "Can't parse time".to_string()).with_source(e)
```
Comment (Xuanwo) [inline crates/iceberg/src/spec/values.rs:267]: The same.

File: crates/iceberg/src/spec/values.rs [hunk 3]
```diff
@@ -86,6 +88,349 @@ pub enum Literal {
...
+    /// use uuid::Uuid;
+    /// use iceberg::spec::values::Literal;
+    /// let t1 = Literal::uuid_from_str("a1a2a3a4-b1b2-c1c2-d1d2-d3d4d5d6d7d8").unwrap();
+    /// let t2 = Literal::uuid(Uuid::from_u128_le(0xd8d7d6d5d4d3d2d1c2c1b2b1a4a3a2a1));
+    ///
+    /// assert_eq!(t1, t2);
+    /// ```
+    pub fn uuid_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
+        let uuid = Uuid::parse_str(s.as_ref())
+            .map_err(|e| Error::new(ErrorKind::DataInvalid, "Can't parse uuid.").with_source(e))?;
```
Comment (Xuanwo) [inline crates/iceberg/src/spec/values.rs:375]: The same.

**Review by Xuanwo — APPROVED** [review]
Thanks a lot!

**Review by JanKaul — APPROVED** [review]
Great work, thank you!

**Comment by liurenjie1024** [comment]
CC @Fokko PTAL

**Review by Fokko — APPROVED** [review]
Great work @liurenjie1024 Looking good!

**Pull request merged**